src/components/TextInput/TextInput.test.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import {
+  render,
+  fireEvent,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import TextInput from './TextInput';
+
+describe('TextInput', () => {
+  /**
+   * Props Validation
+   */
+  test('Throws error if required prop "value" is not supplied to component', () => {
+    console.error = jest.fn(); // eslint-disable-line no-console
+    render(<TextInput onChange={() => null} />);
+    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
+    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
+      .toContain('Failed prop type: The prop `value`');
+  });
+
+  test('Throws error if required prop "onChange" is not supplied to component', () => {
+    console.error = jest.fn(); // eslint-disable-line no-console
+    render(<TextInput value="hello" />);
+    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
+    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
+      .toContain('Failed prop type: The prop `onChange`');
+  });
+
+  test('Throws an error if "type" prop is anything other than allowed values', () => {
+    console.error = jest.fn(); // eslint-disable-line no-console
+    render(<TextInput onChange={() => null} value="hello" type="notOnTheList" />);
+    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
+    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
+      .toContain('Failed prop type: Invalid prop `type`');
+  });
+
+  /**
+   * Component Functionality
+   */
+  test('onChange event fires callback function', () => {
+    const onChangeMock = jest.fn(() => null);
+
+    render(
+      <TextInput
+        name="firstName"
+        id="firstName"
+        label="first name"
+        value="hello"
+        onChange={onChangeMock}
+      />,
+    );
+    const inputElement = screen.getByDisplayValue('hello');
+
+    fireEvent.change(inputElement, { target: { value: 'good bye' } });
+    expect(onChangeMock).toHaveBeenCalledTimes(1);
```

Comment:
should you check if it was called with the value?

Reply:
@ericyliu Interestingly this seems to be the recommended approach with RTL. I actually tried checking the value but my synthetic events all come back with target of null. Not sure if this is a quirk of RTL or what but we can definitely keep investigating.